compiler/src/iree/compiler/GlobalOptimization/PropagateLinalgTranspose.cpp

```diff
@@ -335,19 +565,23 @@ class SinkTransposeThroughExpandShape
 // and isn't the way this pass is modeled. Global data layout transformations
 // like that are better suited for pack/unpack propagation rooted on specific
 // operations.
-class FuseTransposeWithGenericConsumer
-    : public OpRewritePattern<linalg::GenericOp> {
+class FuseTransposeWithLinalgOpConsumer
```

Comment:
I think this too might be simpler if you just generalize the transpose and just call the upstream fusion transformation.

Reply:
I'm getting some confusing rewriter errors trying to do this generalization right now. I'm thinking it will be easier to change upstream to work off of linalg ops and then circle back and change this pattern.

Reply:
Added a TODO for it, marking this as resolved.